pkg/config/remote/service/clients.go

```diff
@@ -0,0 +1,59 @@
+package service
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/benbjohnson/clock"
+)
+
+type client struct {
+	expireAt time.Time
+	pbClient *pbgo.Client
+}
+
+func (c *client) expired(clock clock.Clock) bool {
+	return clock.Now().After(c.expireAt)
+}
+
+type clients struct {
+	sync.Mutex
```

Comment:
as before, do we really want to expose the internal Lock/Unlock state?

Reply:
same, I'll make the change 👍🏻